### ISSUE TITLE
Doc Review: Authorize SSH Access to the VCH Endpoint VM

### ISSUE THREAD

**Issue by vburenin** [vmware/vic#3371]
I would rephrase "IMPORTANT" section saying that changes do not persist on appliance, thus after reboot enabled SSH access will be lost as well as PASSWORD and SSH KEY will not be stored.

**Comment by hickeng** [comment]
I suggest changing the topic to be:

# Enable shell access to the VCH Endpoint VM #

By default all shell access to the VCH endpoint VM is disabled, and login shells for all users are set to `/bin/false`. The command line utility for vSphere Integrated Containers Engine, `vic-machine`, provides a `debug` command that allows you to enable shell access to the VCH endpoint VM, via the VM console or via SSH. All changes made by `vic-machine debug` are non-persistent and will be discarded should the endpointVM be rebooted.

Additionally, deploying a VCH with a debug value of 3 or higher will enable console access by default with a root password of `password`. This is intended to allow for targeted interactive diagnostics in environments where an endpointVM failure occurs consistently and in a fashion that prevents `vic-machine debug` from functioning. _There is no provision for persistently changing this password and this configuration should not be used other than for debugging in a secured environment_

1. On the system on which you run `vic-machine`, navigate to the directory that contains the `vic-machine` utility.
2. Run the `vic-machine debug` command. 
    - You must provide the vSphere target, credentials, and potentially the certificate thumbprint as with all `vic-machine` commands
      - the credentials provided must have the `Virtual machine.Guest Operations.Guest Operation Program Execution` privilege on the endpointVM
    - Supply a password for the root user via the `--rootpw` option to enable just access via the VM console, or see the next section for enabling SSH access
      - Be careful with quoting or escaping shell characters such as `$`, `!` or `%` - wrap the password in single quotes (Linux or Mac OS) or double quotes (Windows) if it includes special characters.

For descriptions of the options that `vic-machine debug` includes in addition to the [Common `vic-machine` Options](common_vic_options.md) , see [VCH Debug Options](debug_vch_options.md).


## Authorize SSH Access to the VCH Endpoint VM ##

Use the same `vic-machine debug` command as above, with the addition of the `--enable-ssh` option.
  - Optionally, specify the `--authorized-key` option to upload a public key file to `/root/.ssh/authorized_keys` folder in the endpoint VM. Include the name of the `*.pub` file in the path.

## Example ##

<pre>$ vic-machine-<i>operating_system</i> debug
--target <i>esxi_host_address</i>
--user root
--password <i>esxi_host_password</i>
--thumbprint <i>certificate_thumbprint</i>
--enable-ssh
--rootpw '<i>new_p@ssword</i>'
--authorized-key <i>path_to_public_key_file</i>/<i>key_file</i>.pub
</pre>

**Result**

The output of the `vic-machine debug` command includes confirmation that SSH access is enabled:

<pre>### Configuring VCH for debug ####
[...]
SSH to appliance:
ssh root@<i>vch_address</i>
[...]
Completed successfully</pre>

**Comment by stuclem** [comment]
Thanks @vburenin and @hickeng.

@hickeng I took all of your proposed content, but since are are talking about 3 different things here, I split them into 3 places:

- Added the info about [`vic-machine create --debug`](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_installation/vch_installer_options.md#--debug) to that section in the big options topic.
- Created [Enable shell access to the VCH Endpoint VM](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_admin/vch_shell_access.md) and [Authorize SSH Access to the VCH Endpoint VM](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_admin/vch_ssh_access.md) as separate topics.
- Added a new wrapper topic, [Debugging the VCH](https://github.com/stuclem/vic/blob/master/doc/user_doc/vic_admin/debug_vch.md), that serves to introduce the two topics on console and SSH access, and that clarifies the difference between `vic-machine debug` and `vic-machine create --debug`.
- Moved the 3 topics out of the VCH Admin section and into the Troubleshooting section. 

However, I just tested both options with build 7313 (i.e. enabling console access only by setting `--root-pw` on one VCH, and enabling SSH access by setting `--root-pw` and `--enable-ssh` on another VCH) and as far as I can tell the result is identical. I was able to log into the VCH endpoint VM by using SSH in both cases.

Changes are in https://github.com/stuclem/vic/commit/3848087af497f05fcc9c60d02cf1f27d36ab3ff3.

This definitely needs another pass.

**Comment by vburenin** [comment]
LGTM @stuclem

**Comment by hickeng** [comment]
@stuclem you're correct, at the moment ssh is enabled irrespective of the presence of `--enable-ssh` - that's a bug.

**Comment by stuclem** [comment]
Thanks for the LGTMs @vburenin and @hickeng. Should we release note the bug?
